Framework/src/Checker.cxx

```diff
@@ -70,120 +114,93 @@ Checker::~Checker()
 
 void Checker::init(framework::InitContext&)
 {
-  // configuration
-  try {
-    std::unique_ptr<ConfigurationInterface> config = ConfigurationFactory::getConfiguration(mConfigurationSource);
-    // configuration of the database
-    mDatabase = DatabaseFactory::create(config->get<std::string>("qc.config.database.implementation"));
-    mDatabase->connect(config->getRecursiveMap("qc.config.database"));
-    LOG(INFO) << "Database that is going to be used : ";
-    LOG(INFO) << ">> Implementation : " << config->get<std::string>("qc.config.database.implementation");
-    LOG(INFO) << ">> Host : " << config->get<std::string>("qc.config.database.host");
-  } catch (
-    std::string const& e) { // we have to catch here to print the exception because the device will make it disappear
-    LOG(ERROR) << "exception : " << e;
-    throw;
-  } catch (...) {
-    std::string diagnostic = boost::current_exception_diagnostic_information();
-    LOG(ERROR) << "Unexpected exception, diagnostic information follows:\n"
-               << diagnostic;
-    throw;
+  initDatabase();
+  initMonitoring();
+  for (auto& check: mChecks){
+    check.init();
   }
-
-  // monitoring
-  try {
-    mCollector = MonitoringFactory::Get("infologger://");
-  } catch (...) {
-    std::string diagnostic = boost::current_exception_diagnostic_information();
-    LOG(ERROR) << "Unexpected exception, diagnostic information follows:\n"
-               << diagnostic;
-    throw;
-  }
-  startFirstObject = system_clock::time_point::min();
-  timer.reset(1000000); // 10 s.
 }
 
 void Checker::run(framework::ProcessingContext& ctx)
 {
-  mLogger << "Receiving " << ctx.inputs().size() << " MonitorObjects" << AliceO2::InfoLogger::InfoLogger::endm;
 
   // Save time of first object
   if (startFirstObject == std::chrono::system_clock::time_point::min()) {
     startFirstObject = system_clock::now();
   }
 
-  std::shared_ptr<TObjArray> moArray{ framework::DataRefUtils::as<TObjArray>(*ctx.inputs().begin()) };
-  moArray->SetOwner(false);
-  auto checkedMoArray = std::make_unique<TObjArray>();
-  checkedMoArray->SetOwner();
-
-  for (const auto& to : *moArray) {
-    std::shared_ptr<MonitorObject> mo{ dynamic_cast<MonitorObject*>(to) };
-    moArray->RemoveFirst();
-    if (mo) {
-      check(mo);
-      store(mo);
-      mTotalNumberHistosReceived++;
-      checkedMoArray->Add(new MonitorObject(*mo));
-    } else {
-      mLogger << "the mo is null" << AliceO2::InfoLogger::InfoLogger::endm;
+  for (const auto& input: mInputs){
+    auto dataRef = ctx.inputs().get(input.binding.c_str());
+    if(dataRef.header != nullptr && dataRef.payload != nullptr){
+      auto moArray = ctx.inputs().get<TObjArray*>(input.binding.c_str());
+      mLogger << "Device " << mDeviceName
+              << " received " << moArray->GetSize()
```

Comment:
GetEntries()
This is just ROOT being ROOT - GetSize() gives you the allocated amount of space for entries, not the number of valid entries.